lib/helpers.js

```diff
@@ -47,6 +57,72 @@ function fetchAndCache(request, options) {
   });
 }
 
+var staticIdb;
+function idbPromise() {
+  if (staticIdb) {
+    return Promise.resolve(staticIdb);
+  }
+
+  return simpleDB.open('sw-toolbox-lru-order').then(function(idb) {
+    staticIdb = idb;
+    return staticIdb;
+  });
+}
+
+var cacheExpirationPromiseChain;
+function queueCacheExpiration(request, cache, options) {
+  var cacheExpiration = cacheExpirationPromiseFactory.bind(null, request, cache, options);
+
+  if (cacheExpirationPromiseChain) {
+    cacheExpirationPromiseChain = cacheExpirationPromiseChain.then(cacheExpiration);
+  } else {
+    cacheExpirationPromiseChain = cacheExpiration();
+  }
+}
+
+function cacheExpirationPromiseFactory(request, cache, options) {
```

Comment:
Possible simplification:
- store a 'last-used' timestamp with each URL
- do an update-or-insert for the URL we just cached to set the timestamp to now
- use IDBIndex.count() to check whether we are over quota
- use a cursor on the index to get the correct number of least-recently-used items and delete them.

Actually, now that I write it out I'm not sure if it really is simpler. It just seems that fetching the whole data set as an array and manipulating that is inefficient somehow.

Bonus: Using a timestamp with an index rather than insert order means that we get time-based cache expiration almost for free.